packages/cli/src/utils.ts

```diff
@@ -130,150 +130,71 @@ export const validFieldsFromPkg = {
-    let output: Record<string, ExportsConditions> = {};
-    pkg.entrypoints.forEach((entrypoint) => {
-      if (entrypointName === entrypoint.name) {
-        let obj: ExportsConditions = exportsHelpers.root(
-          pkg,
-          hasModuleBuild,
-          entrypointName
-        );
-        if (hasBrowserField) {
-          obj = {
-            browser: exportsHelpers.target(
-              pkg,
-              hasModuleBuild,
-              entrypointName,
-              "browser"
-            ),
-            ...obj,
-          };
-        }
-        if (hasWorkerField) {
-          obj = {
-            worker: exportsHelpers.target(
-              pkg,
-              hasModuleBuild,
-              entrypointName,
-              "worker"
-            ),
-            ...obj,
-          };
-        }
-        output["."] = obj;
-      } else {
-        const entrypointPath = nodePath
-          .relative(pkg.directory, entrypoint.source)
-          .replace("src/", "")
-          .replace(/\.[tj]sx?$/, "");
-
-        let conditions: ExportsConditions = exportsHelpers.root(
-          pkg,
-          entrypoint.json.module !== undefined,
-          entrypoint.name,
-          entrypointPath + "/"
-        );
-
-        if (hasBrowserField) {
-          conditions = {
-            browser: exportsHelpers.target(
-              pkg,
-              hasModuleBuild,
-              entrypoint.name,
-              "browser",
-              entrypointPath + "/"
-            ),
-            ...conditions,
-          };
-        }
-        if (hasWorkerField) {
-          conditions = {
-            worker: exportsHelpers.target(
-              pkg,
-              hasModuleBuild,
-              entrypoint.name,
-              "worker",
-              entrypointPath + "/"
-            ),
-            ...conditions,
-          };
-        }
-        output[`./${entrypointPath}`] = conditions;
-      }
-    });
-    return {
-      "./package.json": "./package.json",
-      ...output,
-      ...pkg.json.preconstruct.exports?.extra,
-    };
-  },
 };
 
-const exportsHelpers = {
-  root(
-    pkg: Package,
-    hasModuleBuild: boolean,
-    entrypointName: string,
-    prefix: string = ""
-  ) {
-    return exportsHelpers.target(
-      pkg,
-      hasModuleBuild,
-      entrypointName,
-      "",
-      prefix
-    );
-  },
-  target(
-    pkg: Package,
-    hasModuleBuild: boolean,
-    entrypointName: string,
-    target: string = "",
-    prefix: string = ""
-  ) {
-    return exportsHelpers.env(
-      pkg,
+export function exportsField(
+  pkg: Package
+): Record<string, ExportsConditions | string> | undefined {
+  if (!pkg.project.experimentalFlags.exports) {
+    return;
+  }
+  if (!pkg.json.preconstruct.exports) {
+    return;
+  }
+  const specifiedConditions: ("worker" | "browser" | "module")[] =
+    (pkg.json.preconstruct.exports?.conditions as any) ?? [];
+  let hasModuleBuild = specifiedConditions.includes("module");
```

Comment:
q: should `"module"` be optional? :thinking_face: What are the downsides of having this enabled by default?

Reply:
I've made `module` required when using the exports field